### ISSUE TITLE
Adding "pause" capabilities to Refreshers

### ISSUE THREAD

**Issue by CRImier** [ZeroPhone/ZPUI#56]
While working on the "update from git" app, I've encountered a usecase where it's needed to print a message on the screen, then continue redrawing the progressbar. For now, I'm using `stop()` and `run_in_background`, but it's a hack that's prone to race conditions. It'd be great to have `pause` and `resume` capabilities to Refresher, so that it could be temporarily stopped. I'm guessing that `is_foreground` property could be used as a flag, especially given that many Refresher's functions are wrapped by `to_be_foreground`.

**Comment by CRImier** [comment]
Once the capability is in place, it has to be used in the update app, replacing the `stop()` and `run_in_background()` calls.

**Comment by CRImier** [comment]
Has been merged into master - closing
